simulation/src/gazebo_sensor_plugin_depth_cam/gazebo_sensor_plugin_depth_cam.cc

```diff
@@ -119,63 +113,67 @@ class GazeboSensorPluginHazCam : public FreeFlyerSensorPlugin {
     point_cloud_msg_.point_step = sizeof(float) * num_channels_;
 
     // Declare the striped memory layout.
-    sensor_msgs::PointField field;
+    sensor_msgs::msg::PointField field;
     field.name = "x";
     field.offset = 0 * sizeof(float);
-    field.datatype = sensor_msgs::PointField::FLOAT32;
+    field.datatype = sensor_msgs::msg::PointField::FLOAT32;
     field.count = 1;  // Number of ELEMENTS, not bytes
     point_cloud_msg_.fields.push_back(field);
     field.name = "y";
     field.offset = 1 * sizeof(float);
-    field.datatype = sensor_msgs::PointField::FLOAT32;
+    field.datatype = sensor_msgs::msg::PointField::FLOAT32;
     field.count = 1;  // Number of ELEMENTS, not bytes
     point_cloud_msg_.fields.push_back(field);
     field.name = "z";
     field.offset = 2 * sizeof(float);
-    field.datatype = sensor_msgs::PointField::FLOAT32;
+    field.datatype = sensor_msgs::msg::PointField::FLOAT32;
     field.count = 1;  // Number of ELEMENTS, not bytes
     point_cloud_msg_.fields.push_back(field);
     field.name = "intensity";  // for voxblox
     field.offset = 3 * sizeof(float);
-    field.datatype = sensor_msgs::PointField::FLOAT32;
+    field.datatype = sensor_msgs::msg::PointField::FLOAT32;
     field.count = 1;  // Number of ELEMENTS, not bytes
     point_cloud_msg_.fields.push_back(field);
 
     // Read configuration
     config_reader::ConfigReader config;
     config.AddFile("simulation/simulation.config");
     if (!config.ReadFiles()) {
-      ROS_FATAL("Failed to read simulation config file.");
+      FF_FATAL("Failed to read simulation config file.");
       return;
     }
     bool dos = true;
     if (!config.GetBool("disable_cameras_on_speedup", &dos))
-      ROS_FATAL("Could not read the drawing_width parameter.");
-    if (!config.GetReal("haz_cam_rate", &rate_))
-      ROS_FATAL("Could not read the drawing_width parameter.");
+      FF_FATAL("Could not read the drawing_width parameter.");
+    if (!config.GetReal(is_haz_cam_ ? "haz_cam_rate" : "perch_cam_rate", &rate_))
+      FF_FATAL("Could not read the camera rate parameter.");
     config.Close();
 
     // If we have a sped up simulation and we need to disable the camera
     double simulation_speed = 1.0;
-    if (nh->getParam("/simulation_speed", simulation_speed))
+    if (nh->get_parameter("/simulation_speed", simulation_speed))
       if (simulation_speed > 1.0 && dos) rate_ = 0.0;
   }
 
   // Only send measurements when extrinsics are available
-  void OnExtrinsicsReceived(ros::NodeHandle *nh) {
+  void OnExtrinsicsReceived(NodeHandle& nh) {
     // Toggle if the camera is active or not
     ToggleCallback();
+    // no callback for subscribers, check for subscribers on timer
+    timer_.createTimer(0.5,
```

Comment:
I'm glad you went with this solution instead of just publishing the image all the time.

Reply:
@bcoltin I came here to see how long you put the timer, such that I can put the same in the nav_cam and haz_cam plugin :)